pkg/agent/openflow/pipeline.go

```diff
@@ -1901,6 +2101,78 @@ func (c *client) snatRuleFlow(ofPort uint32, snatIP net.IP, snatMark uint32, loc
 	}
 }
 
+func (c *client) serviceClassifierFlow(groupID binding.GroupIDType, svcIP net.IP, svcPort uint16, protocol binding.Protocol,
+	affinityTimeout uint16, nodeLocalExternal bool) []binding.Flow {
+	idleTimeout := serviceDstMacRewriteIdleTimeOut
+	hardTimeout := uint16(0)
+	// If the SessionAffinity of Service is set, the learned flow should be set with hard timeout. Assumed that the SessionAffinity
+	// is 300s, if the learned flow is set with idle timeout 60s, the learned flow may be deleted when reach idle timeout is reached,
+	// then the response traffic cannot be returned correctly.
+	if affinityTimeout != 0 {
+		idleTimeout = 0
+		hardTimeout = affinityTimeout
+	}
+
+	// Use unique cookie ID here to avoid learned flow cascade deletion.
+	cookieID := c.cookieAllocator.RequestWithObjectID(cookie.Service, uint32(groupID)).Raw()
+	// These flows are used to match request NodePort/LoadBalancer traffic from outside the cluster and generate a flow that rewrites
```

Comment:
Probably remove "request"

Reply:
What does "generate a flow" mean?

Reply:
These flows are used to match the first packet of NodePort/LoadBalancer traffic from outside the cluster. One action is that generate a learned flow which rewrites destination MAC address of response packets and sets the output port.

Reply:
> What does "generate a flow" mean?

This means that current flow can generate a learned flow.